maxrefdes178-DigitDetection/maxrefdes178_max78000_video/src/max78000_video_main.c

```diff
@@ -0,0 +1,1214 @@
+        }
+
+        for (ch = 0; ch < (NUM_CLASSES); ++ch) {
+            sum += exp(prior_cls[i * NUM_CLASSES + ch] / 128.);
+        }
+
+        for (ch = 0; ch < (NUM_CLASSES); ++ch) {
+            prior_cls_softmax[i * NUM_CLASSES + ch] =
+                (uint16_t)(65536. * exp(prior_cls[i * NUM_CLASSES + ch] / 128.) / sum);
+        }
+    }
+}
+
+static void get_prior_locs(void)
+{
+    int8_t* loc_addr = (int8_t*)0x50403000;
+
+    int ar_idx, scale_idx, rel_idx, prior_idx, prior_count;
+
+    for (ar_idx = 0; ar_idx < NUM_ARS; ++ar_idx) {
+        int8_t* loc_addr_temp = loc_addr;
+
+        for (scale_idx = 0; scale_idx < NUM_SCALES; ++scale_idx) {
+            prior_count = SQUARE(dims[scale_idx]);
+
+            for (rel_idx = 0; rel_idx < prior_count; ++rel_idx) {
+                prior_idx = get_prior_idx(ar_idx, scale_idx, rel_idx);
+                memcpy(&prior_locs[LOC_DIM * prior_idx], loc_addr_temp, LOC_DIM);
+                loc_addr_temp += LOC_DIM;
+            }
+        }
+
+        loc_addr += 0x8000;
+    }
+}
+
+static void get_prior_cls(void)
+{
+    int8_t* cl_addr = (int8_t*)0x50803000;
+
+    int ar_idx, cl_idx, scale_idx, rel_idx, prior_idx, prior_count;
+
+    for (ar_idx = 0; ar_idx < NUM_ARS; ++ar_idx) {
+        for (cl_idx = 0; cl_idx < NUM_CLASSES; cl_idx += 4) {
+            int8_t* cl_addr_temp = cl_addr;
+
+            for (scale_idx = 0; scale_idx < NUM_SCALES; ++scale_idx) {
+                prior_count = SQUARE(dims[scale_idx]);
+
+                for (rel_idx = 0; rel_idx < prior_count; ++rel_idx) {
+                    prior_idx = get_prior_idx(ar_idx, scale_idx, rel_idx);
+                    memcpy(&prior_cls[NUM_CLASSES * prior_idx + cl_idx], cl_addr_temp, 4);
+                    cl_addr_temp += 4;
+                }
+            }
+
+            cl_addr += 0x8000;
+
+            if ((cl_addr == (int8_t*)0x50823000) || (cl_addr == (int8_t*)0x50c23000)) {
+                cl_addr += 0x003e0000;
+            }
+        }
+    }
+
+    softmax();
+}
+
+static void get_priors(void)
+{
+    get_prior_locs();
+    get_prior_cls();
+}
+
+static float calculate_IOU(float* box1, float* box2)
+{
+    float x_left   = MAX(box1[0], box2[0]);
+    float y_top    = MAX(box1[1], box2[1]);
+    float x_right  = MIN(box1[2], box2[2]);
+    float y_bottom = MIN(box1[3], box2[3]);
+    float intersection_area;
+
+    if (x_right < x_left || y_bottom < y_top) {
+        return 0.0;
+    }
+
+    intersection_area = (x_right - x_left) * (y_bottom - y_top);
+
+    float box1_area = (box1[2] - box1[0]) * (box1[3] - box1[1]);
+    float box2_area = (box2[2] - box2[0]) * (box2[3] - box2[1]);
+
+    float iou = (float)(intersection_area) / (float)(box1_area + box2_area - intersection_area);
+
+    return iou;
+}
+
+static void get_cxcy(float* cxcy, int prior_idx)
+{
+    int i, scale_idx, ar_idx, rel_idx, cx, cy;
+
+    get_indices(&ar_idx, &scale_idx, &rel_idx, prior_idx);
+
+    cy      = rel_idx / dims[scale_idx];
```

Comment:
It throws waring:
src/max78000_video_main.c: In function 'get_cxcy':
src/max78000_video_main.c:1025:29: warning: 'scale_idx' may be used uninitialized in this function [-Wmaybe-uninitialized]
 1025 |     cy      = rel_idx / dims[scale_idx];